table/queries/merged_reduced_scans.sql

```diff
@@ -21,6 +21,87 @@ CREATE TEMP FUNCTION CleanError(error STRING) AS (
+    WHEN ENDS_WITH(error, "protocol not available") THEN "setup/invalid" # ipv4 vs 6 error
+    WHEN ENDS_WITH(error, "too many open files") THEN "setup/invalid"
+
+    # Dial failures
+    WHEN ENDS_WITH(error, "network is unreachable") THEN "dial/tcp.address_unreachable"
+    WHEN ENDS_WITH(error, "no route to host") THEN "dial/tcp.no_route"
+    WHEN ENDS_WITH(error, "connection refused") THEN "dial/tcp.refused"
+    WHEN ENDS_WITH(error, "context deadline exceeded") THEN "dial/tcp.timed_out"
+    WHEN ENDS_WITH(error, "connect: connection timed out") THEN "dial/tcp.timed_out"
+    WHEN REGEXP_CONTAINS(error, "^connection reset by peer") THEN "dial/tcp.reset" #no read: or write: prefix in error
+    WHEN ENDS_WITH(error, "connect: connection reset by peer") THEN "dial/tcp.reset"
+    WHEN ENDS_WITH(error, "getsockopt: connection reset by peer") THEN "dial/tcp.reset"
+
+    # TLS failures
+    WHEN REGEXP_CONTAINS(error, "tls:") THEN "tls/tls.protocol.error"
+    WHEN REGEXP_CONTAINS(error, "remote error:") THEN "tls/tls.failed"
+    WHEN REGEXP_CONTAINS(error, "local error:") THEN "tls/tls.failed"
+    WHEN ENDS_WITH(error, "readLoopPeekFailLocked: <nil>") THEN "tls/tls.failed"
+    WHEN ENDS_WITH(error, "missing ServerKeyExchange message") THEN "tls/tls.protocol.error"
+    WHEN ENDS_WITH(error, "no mutual cipher suite") THEN "tls/tls.protocol.error"
+
+    # Write failures
+    WHEN ENDS_WITH(error, "write: connection reset by peer") THEN "write/tcp.reset"
+    WHEN ENDS_WITH(error, "write: broken pipe") THEN "write/invalid"
+
+    # Read failures
+    WHEN REGEXP_CONTAINS(error, "request canceled") THEN "read/tcp.timed_out"
+    WHEN ENDS_WITH(error, "i/o timeout") THEN "read/tcp.timed_out"
+    WHEN ENDS_WITH(error, "shutdown: transport endpoint is not connected") THEN "read/invalid"
+    # TODO: for HTTPS this error could potentially also be SNI blocking in the tls stage
+    # find a way to diffentiate this case.
+    WHEN ENDS_WITH(error, "read: connection reset by peer") THEN "read/tcp.reset" 
+
+    # HTTP content verification failures
+    WHEN (source != "ECHO" AND REGEXP_CONTAINS(error, "EOF")) THEN "http/tcp.closed"
+    WHEN ENDS_WITH(error, "http: server closed idle connection") THEN "http/tcp.closed"
+    WHEN ENDS_WITH(error, "trailer header without chunked transfer encoding") THEN "http/http.response.invalid"
+    WHEN ENDS_WITH(error, "response missing Location header") THEN "http/http.response.invalid"
+    WHEN REGEXP_CONTAINS(error, "bad Content-Length") THEN "http/http.response.invalid"
+    WHEN REGEXP_CONTAINS(error, "failed to parse Location header") THEN "http/http.response.invalid"
+    WHEN REGEXP_CONTAINS(error, "malformed HTTP") THEN "http/http.response.invalid"
+    WHEN REGEXP_CONTAINS(error, "malformed MIME") THEN "http/http.response.invalid"
+
+    # Content verification failures
+    WHEN (source = "ECHO" AND ENDS_WITH(error, "EOF")) THEN "content/tcp.closed"
```

Comment:
This is not a TCP error. It's an valid TCP session, with empty content. Perhaps `content/empty`?

Reply:
switched this to "content/response_mismatch"